src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -381,47 +394,132 @@ private void DropItems(Character killer)
+
+					if (killer == null || dropChance > autoloot)
+					{
+						var direction = new Direction(rnd.Next(0, 360));
+						var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
+						var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+
+						dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
+						dropItem.Drop(this.Map, this.Position, direction, distance);
+					}
+					else
+					{
+						killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
+					}
 				}
+			}
+		}
 
-				var dropItem = new Item(itemData.Id);
-				var minAmount = dropItemData.MinAmount;
-				var maxAmount = dropItemData.MaxAmount;
+		/// <summary>
+		/// Rolls for a chance for this monster to become a rare (jackpot or elite) monster
+		/// </summary>
+		/// <param name="redOrb">true if this was spawned by a red orb and should get the boosted rates</param>
+		/// <returns></returns>
+		public void PossiblyBecomeRare(bool redOrb = false)
+		{
+			var rnd = RandomProvider.Get();
 
-				if (dropItemData.ItemId == ItemId.Silver || dropItemData.ItemId == ItemId.Gold)
-				{
-					minAmount = Math.Max(1, (int)(minAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-					maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-				}
+			var worldConf = ZoneServer.Instance.Conf.World;
 
-				dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+			var spawnRateBoost = 1F;
+			if (redOrb)
+				spawnRateBoost = worldConf.RedOrbJackpotRate / 100F;
 
-				if (killer == null || dropChance > autoloot)
-				{
-					var direction = new Direction(rnd.Next(0, 360));
-					var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
-					var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+			if (rnd.NextDouble() * 100D < worldConf.GoldJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.TwinkleBuff, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
 
-					dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
-					dropItem.Drop(this.Map, this.Position, direction, distance);
-				}
-				else
-				{
-					killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
-				}
+			if (rnd.NextDouble() * 100D < worldConf.BlueJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperExp, 1, 0, TimeSpan.Zero, this);
+				return;
 			}
+
+			if (rnd.NextDouble() * 100D < worldConf.SilverJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperDrop, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
+
+			// Check to see if this monster can become elite
+			// Note that on official it uses the map level rather than the monster level
+			if (this.Level < worldConf.EliteMinLevel)
+				return;
+
+			if (redOrb)
+				spawnRateBoost = worldConf.RedOrbEliteRate / 100F;
+
+			if (rnd.NextDouble() * 100D < worldConf.EliteSpawnChance * spawnRateBoost)
```

Comment:
The double notation (which should use a lower-case d) isn't necessary and could safely be omitted since it doesn't contribute anything to the code or the reader's ability to understand it.

Reply:
Fixed.